src/components/Create/IncentivesForm.tsx

```diff
@@ -30,7 +30,7 @@ export default function IncentivesForm({
   useLayoutEffect(() => {
     setDiscountRate(initialDiscountRate)
     setBondingCurveRate(initialBondingCurveRate)
-  }, [])
+  }, [initialBondingCurveRate, initialDiscountRate])
```

Comment:
some of these dependency arrays were empty to ensure they only run once when the component loads. i'm not sure if a better practice there is to just omit the second argument to useEffect entirely. 

have you tested to see if adding dependencies breaks anything?

Reply:
Using useEffect without the dependency array will cause it to run on every render which is probably not what we want in these cases ([source](https://reactjs.org/docs/hooks-reference.html#useeffect)).

From my experience, I've found that not using exhaustive-deps typically is a sign that useEffect is not being used properly:
![image](https://user-images.githubusercontent.com/238271/142973270-af9a8753-b613-41c9-855d-ca00c8e074a0.png)

The FAQ [here](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) describes a bunch of reasons why deps should be exhaustive and why exhaustive-deps is a lint rule.

I manually updated each of the dependencies that the linter caught. You probably noticed some of the refactoring tha had to happen as a result.